tools/otci/otci/otci.py

```diff
@@ -2619,6 +2638,165 @@ def wait_for(self, command: str, expect_line: Optional[Union[str, Pattern, Colle
+                      transmit_time: int = 10,
+                      length: Optional[int] = None) -> Dict[str, Dict[str, Any]]:
+        """Run iperf3 in client mode.
+
+        :param host: The host IPv6 address to send iperf3 traffic.
+        :param ipv6: True to use IPv6, FALSE to use IPv4 (default IPv6).
+        :param udp: True to use UDP, FALSE to use TCP (default UDP).
+        :param bind_address: The local address to be bound.
+        :param bitrate: The target bitrate in bits/sec (default 10000 bit/sec).
+        :param interval: Seconds between periodic throughput reports (default 1 sec).
+        :param transmit_time: Time in seconds to transmit for (default 10 secs)
+        :param length: Length of buffer to read or write (default None).
+        """
+        #
+        # Iperf3 client example:
+        #
+        # $ iperf3 -6 -c fd6e:debe:d06c:f110:7903:1c0a:854d:a991 -u -b 90k  -t 5 -l 1232
+        # Connecting to host fd6e:debe:d06c:f110:7903:1c0a:854d:a991, port 5201
+        # [  5] local fdb5:1c54:4233:1:f316:192e:626b:171f port 33577 connected to fd6e:debe:d06c:f110:7903:1c0a:854d:a991 port 5201
+        # [ ID] Interval           Transfer     Bitrate         Total Datagrams
+        # [  5]   0.00-1.00   sec  12.0 KBytes  98.6 Kbits/sec  10
+        # [  5]   1.00-2.00   sec  10.8 KBytes  88.7 Kbits/sec  9
+        # [  5]   2.00-3.00   sec  10.8 KBytes  88.7 Kbits/sec  9
+        # [  5]   3.00-4.00   sec  10.8 KBytes  88.7 Kbits/sec  9
+        # [  5]   4.00-5.00   sec  10.8 KBytes  88.7 Kbits/sec  9
+        # - - - - - - - - - - - - - - - - - - - - - - - - -
+        # [ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
+        # [  5]   0.00-5.00   sec  55.3 KBytes  90.7 Kbits/sec  0.000 ms  0/46 (0%)  sender
+        # [  5]   0.00-6.35   sec  34.9 KBytes  45.0 Kbits/sec  80.014 ms  16/45 (36%)  receiver
+        #
+        # iperf Done.
+
+        wait_time = 10
+        client_option = f'-c {host}'
+        version_option = "-6" if ipv6 else "-4"
+        udp_option = '-u' if udp else ''
+        bind_option = '' if bind_address is None else f'-B {bind_address}'
+        bitrate_option = f'-b {bitrate}'
+        interval_option = f'-i {interval}'
+        time_option = f'-t {transmit_time}'
+        length_option = '' if length is None else f'-l {length}'
+
+        cmd = f'iperf3 {version_option} {client_option} {udp_option} {bitrate_option} {interval_option} {time_option} {length_option}'
+        output = self.execute_platform_command(cmd, timeout=transmit_time + wait_time)
+
+        results = {}
+        for line in output:
+            fields = line.split()
+            if len(fields) == 13:
+                if fields[-1] == 'sender':
+                    results['sender'] = self.__parse_iperf3_report(line)
+                elif fields[-1] == 'receiver':
+                    results['receiver'] = self.__parse_iperf3_report(line)
```

Comment:
```suggestion
            if len(fields) != 13:
                continue
            if fields[-1] == 'sender':
                results['sender'] = self.__parse_iperf3_report(line)
            elif fields[-1] == 'receiver':
                results['receiver'] = self.__parse_iperf3_report(line)
```
nit. The depth of nested ifs could be smaller.